Gems/ROS2/Code/Source/Communication/PublisherConfiguration.cpp

```diff
@@ -30,6 +31,7 @@ namespace ROS2
                         AZ::Edit::UIHandlers::Default, &PublisherConfiguration::m_publish, "Publish", "Whether the publisher is on")
                     ->DataElement(
                         AZ::Edit::UIHandlers::Default, &PublisherConfiguration::m_topicConfiguration, "Topic", "Topic configuration")
+                    ->Attribute(AZ::Edit::Attributes::ChangeValidate, &ROS2Names::ValidateFullTopicField)
```

Comment:
Can we guarantee that this does not impact UX negatively? 
For example, if the namespace is invalid, the validate error will persist whatever the input to PublisherConfiguration, which is a bad experience for the user.

If we guarantee that namespace part will always be valid here, then it is fine.